dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +113,231 @@ public Bundle getBundleById(String id) throws DotDataException {
 	@Override
 	public void deleteBundle(String id) throws DotDataException {
 		bundleFactory.deleteBundle(id);
+	}
+
+	@WrapInTransaction
+	private void internalDeleteBundleAndDependencies (final String bundleId, final User user) throws DotDataException { // todo: make this batcheable
+
+		this.validateBundleDeletePermission(user, bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId +
+				" and dependencies by: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Removing audit status for a bundle: " + bundleId);
+			this.publishAuditAPI.deletePublishAuditStatus(bundleId);
+		} catch (DotPublisherException e) {
+
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsAPI.deletePushedAssetsByBundleId(bundleId);
```

Comment:
Is it safe to keep removing data even if `this.publishAuditAPI.deletePublishAuditStatus(bundleId);` failed?

Reply:
if it fails it will throw the exception at line 132 @jcastro-dotcms

Reply:
Yeah, it is an transaction so if any methods fails, the rest stop and rollback